STL="gnustl_static" compile error for build_android.sh

**android-ndk-r16b**

/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1301:13: error: expected unqualified-id
            CVTGRAYIMPL(3);
            ^
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1260:21: note: expanded from macro 'CVTGRAYIMPL'
    uint8x8x##n##_t _S;                                                 \
                    ^
/home/joe/ndk_vulkan_ncnn/android-ndk-r16b/sources/cxx-stl/gnu-libstdc++/4.9/libs/arm64-v8a/include/bits/ctype_base.h:41:12: note: expanded from macro '_S'
#define _S _CTYPE_S
           ^
/home/joe/ndk_vulkan_ncnn/android-ndk-r16b/sysroot/usr/include/ctype.h:49:18: note: expanded from macro '_CTYPE_S'
#define _CTYPE_S 0x08
                 ^
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1301:13: error: expression is not assignable
            CVTGRAYIMPL(3);
            ^~~~~~~~~~~~~~
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1261:9: note: expanded from macro 'CVTGRAYIMPL'
    _S  = vld##n##_u8(Sp);                                              \
    ~~  ^
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1301:13: error: member reference base type 'int' is not a structure or union
            CVTGRAYIMPL(3);
            ^~~~~~~~~~~~~~
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1262:22: note: expanded from macro 'CVTGRAYIMPL'
    _Bh = vmovl_u8(_S.val[0]);                                          \
                   ~~^~~~
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1301:13: error: member reference base type 'int' is not a structure or union
            CVTGRAYIMPL(3);
            ^~~~~~~~~~~~~~
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1263:22: note: expanded from macro 'CVTGRAYIMPL'
    _Gh = vmovl_u8(_S.val[1]);                                          \
                   ~~^~~~
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1301:13: error: member reference base type 'int' is not a structure or union
            CVTGRAYIMPL(3);
            ^~~~~~~~~~~~~~
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1264:22: note: expanded from macro 'CVTGRAYIMPL'
    _Rh = vmovl_u8(_S.val[2]);                                          \
                   ~~^~~~
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1303:13: error: expected unqualified-id
            CVTGRAYIMPL(4);
            ^
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1260:21: note: expanded from macro 'CVTGRAYIMPL'
    uint8x8x##n##_t _S;                                                 \
                    ^
/home/joe/ndk_vulkan_ncnn/android-ndk-r16b/sources/cxx-stl/gnu-libstdc++/4.9/libs/arm64-v8a/include/bits/ctype_base.h:41:12: note: expanded from macro '_S'
#define _S _CTYPE_S
           ^
/home/joe/ndk_vulkan_ncnn/android-ndk-r16b/sysroot/usr/include/ctype.h:49:18: note: expanded from macro '_CTYPE_S'
#define _CTYPE_S 0x08
                 ^
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1303:13: error: expression is not assignable
            CVTGRAYIMPL(4);
            ^~~~~~~~~~~~~~
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1261:9: note: expanded from macro 'CVTGRAYIMPL'
    _S  = vld##n##_u8(Sp);                                              \
    ~~  ^
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1303:13: error: member reference base type 'int' is not a structure or union
            CVTGRAYIMPL(4);
            ^~~~~~~~~~~~~~
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1262:22: note: expanded from macro 'CVTGRAYIMPL'
    _Bh = vmovl_u8(_S.val[0]);                                          \
                   ~~^~~~
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1303:13: error: member reference base type 'int' is not a structure or union
            CVTGRAYIMPL(4);
            ^~~~~~~~~~~~~~
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1263:22: note: expanded from macro 'CVTGRAYIMPL'
    _Gh = vmovl_u8(_S.val[1]);                                          \
                   ~~^~~~
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1303:13: error: member reference base type 'int' is not a structure or union
            CVTGRAYIMPL(4);
            ^~~~~~~~~~~~~~
/home/joe/software/TNN/source/tnn/device/arm/arm_util.cc:1264:22: note: expanded from macro 'CVTGRAYIMPL'

--- a/source/tnn/device/arm/arm_util.cc
+++ b/source/tnn/device/arm/arm_util.cc
@@ -1257,11 +1257,11 @@ void NV21ToBGRA(const unsigned char* nv21, unsigned char* bgra, int h, int w) {
 #ifdef TNN_USE_NEON
 
 #define CVTGRAYIMPL(n)                                                  \
-    uint8x8x##n##_t _S;                                                 \
-    _S  = vld##n##_u8(Sp);                                              \
-    _Bh = vmovl_u8(_S.val[0]);                                          \
-    _Gh = vmovl_u8(_S.val[1]);                                          \
-    _Rh = vmovl_u8(_S.val[2]);                                          \
+    uint8x8x##n##_t _Src;                                               \
+    _Src  = vld##n##_u8(Sp);                                            \
+    _Bh   = vmovl_u8(_Src.val[0]);                                      \
+    _Gh   = vmovl_u8(_Src.val[1]);                                      \
+    _Rh   = vmovl_u8(_Src.val[2]);                                      \
     _Bval = vcvtq_f32_u32(vmovl_u16(vget_low_u16(_Bh)));                \
     _Gval = vcvtq_f32_u32(vmovl_u16(vget_low_u16(_Gh)));                \
     _Rval = vcvtq_f32_u32(vmovl_u16(vget_low_u16(_Rh)));                \

--- a/source/tnn/utils/npu_common_utils.cc
+++ b/source/tnn/utils/npu_common_utils.cc
@@ -14,12 +14,11 @@
 
 #include "tnn/utils/npu_common_utils.h"
 
-#include <sstream>
-
 #include "tnn/core/macro.h"
 #include "tnn/interpreter/layer_resource.h"
 #include "tnn/layer/base_layer.h"
 #include "tnn/utils/dims_vector_utils.h"
+#include "tnn/utils/string_utils_inner.h"
 
 namespace TNN_NS {
 
@@ -55,7 +54,7 @@ std::string NpuCommonUtils::GetFileHash(ModelConfig &model_config) {
     int hash                 = 0;
     for (size_t i = 0; i < file_content.length(); ++i)
         hash = 65599 * hash + file_content.at(i);
-    return std::to_string(hash ^ (hash >> 16));
+    return ToString(hash ^ (hash >> 16));
 }
 
 bool NpuCommonUtils::FileExits(std::string model_path) {

--- a/test/test_utils.cc
+++ b/test/test_utils.cc
@@ -108,7 +108,7 @@ int CompareData(const bfp16_t* ref_data, const bfp16_t* result_data, size_t n, f
 
 int CompareData(const int8_t* ref_data, const int8_t* result_data, size_t n) {
     for (unsigned long long i = 0; i < n; i++) {
-        if (fabs(result_data[i] - ref_data[i]) > 1) {
+        if (abs(result_data[i] - ref_data[i]) > 1) {
             LOGE("ERROR AT %llu result %d ref %d\n", i, result_data[i], ref_data[i]);
             return -1;
         }
@@ -122,7 +122,7 @@ int CompareData(const uint8_t* ref_data, const uint8_t* result_data, int mat_cha
         int c = i % mat_channel;
         if (c >= channel)
             continue;
-        if (fabs(result_data[i] - ref_data[i]) > 1) {
+        if (abs(result_data[i] - ref_data[i]) > 1) {
             LOGE("ERROR AT %llu result %d ref %d\n", i, result_data[i], ref_data[i]);
             return -1;
         }

